src/boost_histogram/_internal/hist.py

```diff
@@ -337,6 +338,24 @@ def ndim(self) -> int:
         """
         return self._hist.rank()
 
+    def compare(self, hist2: "Histogram") -> bool:
+        if (np.allclose(self.view().shape, hist2.view().shape)):
+            if (np.allclose(self.view(), hist2.view())):
+                if (np.allclose(self.variances(), hist2.variances())):
+                    if (re.search("(?<=storage=).*", str(self.view))[0].split('(')[0] == re.search("(?<=storage=).*", str(hist2.view))[0].split('(')[0]):
+                        if (list(self.axes)==list(hist2.axes)):
+                            return ""
+                        else:
+                            return "The axes [" + "\033[91m" + str(list(self.axes)) + " and " + str(list(hist2.axes)) + "\033[0m" + "] are not equal."
+                    else:
+                        return "The storage types [" + "\033[91m" + str(re.search("(?<=storage=).*", str(self.view))[0].split('(')[0]) + " and " + str(re.search("(?<=storage=).*", str(hist2.view))[0].split('(')[0]) + "\033[0m" + "] are not equal."
+                else:
+                    return "The histogram contents : \n" + "\033[91m" + str(self.variances()) + "\033[0m" + "\nand\n" + "\033[91m" + str(hist2.variances()) + "\033[0m" + "\nare not equal."
+            else:
+                return "The histogram contents : \n" + "\033[91m" + str(self.view()) + "\033[0m" + "\nand\n" + "\033[91m" + str(hist2.view()) + "\033[0m" + "\nare not equal."
+            else:
```

Comment:
No else after return, please! This would look much cleaner. Also, the final else block is misaligned. (CC @jpivarski this is an example of else after return. :)

I would then add "not" in front of each if, and place the return immediately after. Now you'll have a non-indented linear chain of checks, rather than this sideways pyramid. :)

Reply:
Alright makes sense! Will make the changes :)

Reply:
FYI, what I described is called "guard style". You check for the failure and immediately break the control flow (usually via `return`); everything after that is now guarded from that condition. It keeps logic simpler and reduces indentation.

Ruby actually has syntax for it, `return <value> if <condition>` can be written on one line, and the inverse is `return <value> unless <condition>`. Python never got custom syntax, since `if <condition>: return <value>` isn't actually any longer.

Reply:
Interesting... Thanks for the insight!